src/encoder.rs

```diff
@@ -992,13 +996,16 @@ impl<T: Pixel> FrameInvariants<T> {
     #[cfg(feature = "unstable")]
     if fi.show_frame || fi.showable_frame {
       let cur_frame_time = fi.frame_timestamp();
-      // Increment the film grain seed for the next frame
-      if let Some(params) =
-        Arc::make_mut(&mut fi.config).get_film_grain_mut_at(cur_frame_time)
-      {
-        params.random_seed = params.random_seed.wrapping_add(3248);
-        if params.random_seed == 0 {
-          params.random_seed = DEFAULT_GS_SEED;
+      // SAFETY: We're not inside a tile, there's nothing else writing to the encoder config here.
+      unsafe {
+        // Increment the film grain seed for the next frame
+        if let Some(params) = arc_get_mut_unsafe(&mut fi.config)
```

Comment:
Wouldn't be better to move the film grain information to the FrameInvariants? Also who is accessing the `config` here?